src/Block/AdminListBlockService.php

```diff
@@ -51,13 +51,18 @@ public function __construct(
         Pool $pool,
         ?TemplateRegistryInterface $templateRegistry = null
     ) {
-        parent::__construct($twigOrName, $templating);
+        // NEXT_MAJOR: Remove conditions, only pass the Environment
+        if ($twigOrName instanceof Environment) {
+            parent::__construct($twigOrName);
+        } else {
+            parent::__construct($twigOrName, $templating);
+        }
```

Comment:
Can you make argument 2 optional and add trigger_error like in `AuditBlockService`?